krate/src/main/kotlin/hu/autsoft/krate/Functions.kt

```diff
@@ -46,9 +46,14 @@ public fun Krate.longPref(key: String): ReadWriteProperty<Krate, Long?> {
 
 /**
  * Creates an optional preference of type String with the given [key] in this [Krate] instance.
+ *
+ * @param [key] the key to identify this value in SharedPreferences
+ * @param [isValid] the validation function to be used when setting this property
+ *
+ * @return [StringDelegate]
  */
-public fun Krate.stringPref(key: String): ReadWriteProperty<Krate, String?> {
-    return StringDelegate(key)
+public fun Krate.stringPref(key: String, isValid: (String?) -> Boolean = { true }): ReadWriteProperty<Krate, String?> {
```

Comment:
It has been brought up to me that include a default parameter that's a lambda is expensive. We should favor a function overload, instead.